### ISSUE TITLE
EitherSchema

### ISSUE THREAD

**Issue by eval** [tomdalling/rschema#6]
How would one go about defining a schema that says 'matches x or y'?

I tried predicates but couldn't find a way to re-use existing schemas.

I ended up defining a custom schema:
```ruby
RSchema.schema {{
  some_key: either(String, Array[String])
}}
```

Would this be useful to add?

**Comment by tomdalling** [comment]
Absolutely! I just finished implementing it, and it's available in version 1.3.0.

**Comment by eval** [comment]
Thanks!
